lib/diaspora_federation/entities/reshare.rb

```diff
@@ -31,6 +31,11 @@ def to_s
       # Fetch and receive root post from remote, if not available locally
       # and validates if it's from the correct author
       def validate_root
+        return if root_author.nil? && root_guid.nil?
+
+        raise Entity::ValidationError, "root_guid can't be nil if root_author is present" if root_guid.nil?
+        raise Entity::ValidationError, "root_author can't be nil if root_guid is present" if root_author.nil?
```

Comment:
I would like to have the `#{self}` somewhere in the error message. It's a lot easier to grep the guid when somebody asks "why didn't you receive this message. I already had a some questions from other developers from other software implementing the protocol, why some messages failed. (Even when this error shouldn't happen often)

So maybe just add it to the beginning: `"#{self}: ..."`